Remove apirule v1beta1 dashboard (#2104)
kyma-project/api-gateway: docs/user/troubleshooting-guides/03-80-apirule-v2-introduction/03-83-apirules-are-missing-from-kyma-dashboard.md

# Previously created APIRules are missing from Kyma Dashboard API Rules view

In case you have created APIRules using the `v1beta1` version, and you have not yet migrated them to the `v2` version,
they will not be displayed in the Kyma Dashboard API Rules view.
To display the APIRules in the Kyma Dashboard, you need to migrate them to the `v2` version.

## Symptom

You have created APIRules using the `v1beta1` version, and they are not displayed in the Kyma Dashboard API Rules view.
To check if the APIRules are present in the cluster, you can run the following command:

```bash

> kubectl get apirules.v2.gateway.kyma-project.io -A

Author: I think we should promote the v2 version as it will get confusing when the v1beta1 API will be dropped.

In case you have created APIRules using the `v1beta1` version, and you have not yet migrated them to the `v2` version,
they will not be displayed in the Kyma Dashboard API Rules view.
To display the APIRules in the Kyma Dashboard, you need to migrate them to the `v2` version.

## Symptom

You have created APIRules using the `v1beta1` version, and they are not displayed in the Kyma Dashboard API Rules view.
To check if the APIRules are present in the cluster, you can run the following command:

```bash
kubectl get apirules.v2.gateway.kyma-project.io -A
```
If the APIRules are present, you will see them listed in the output.
However, for APIRules that were not upgraded to the `v2` version, the output will not show the rules field, for example:

```yaml
apiVersion: gateway.kyma-project.io/v2
kind: APIRule
metadata:
  annotations:
    gateway.kyma-project.io/original-version: v1beta1
  name: httpbin
  namespace: test
spec:
    gateway: kyma-system/kyma-gateway
    hosts:
        - httpbin.local.kyma.dev
    service:
        name: httpbin
        namespace: test
        port: 8000
status:
    lastProcessedTime: "2025-04-25T11:16:11Z"
    state: Warning
```

> In case the APIRule was created using the `v1beta1` version, the output will show 
> the `gateway.kyma-project.io/original-version: v1beta1` annotation.

Reviewer: ```suggestion
```
I would move this before the code block

## Cause

The APIRules were originally created using the `v1beta1` version, and you haven't yet migrated them to the `v2` version.
The APIRule v1beta1 API is no longer available either via the Kyma Dashboard or the `kubectl` command.

> To make sure that the support for the APIRule is maintained, you need to migrate the APIRules to the `v2` version.

Reviewer: > To ensure that the APIRule's support is maintained, you need to migrate it to the `v2` version.

I would move this to solution

## Solution

> To migrate the APIRules to the `v2` version, you can follow the [APIRule migration guide](../../apirule-migration/README.md).

Reviewer: okay, makes sense

## Symptom

Kyma dashboard's `API Rules` view does not display APIRules created in version `v1beta1`.
To check if the APIRules are present in the cluster, run the following command:

```bash
kubectl get apirules.v2.gateway.kyma-project.io -A
```
This command lists all APIRules available in your Kyma cluster, regardless of their original version.
To get a specific APIRule and check the version in which it was created, run the following command:
```yaml
kubectl get apirules.v2.gateway.kyma-project.io -n $NAMESPACE $APIRULE_NAME -o yaml
```

If the APIRule was created using version `v1beta1`,
the output contains the annotation `gateway.kyma-project.io/original-version: v1beta1`.

See the following example: 

```yaml
apiVersion: gateway.kyma-project.io/v2
kind: APIRule
metadata:
  annotations:
    gateway.kyma-project.io/original-version: v1beta1
  name: httpbin
  namespace: test
spec:
    gateway: kyma-system/kyma-gateway
    hosts:
        - httpbin.local.kyma.dev
    service:
        name: httpbin
        namespace: test
        port: 8000
status:
    lastProcessedTime: "2025-04-25T11:16:11Z"
    state: Warning
```

## Cause

> The APIRules were originally created using the `v1beta1` version, and you haven't yet migrated them to the `v2` version.

Reviewer: ```suggestion
APIRules that are not displayed in Kyma dashboard were originally created using version `v1beta1`, and you haven't yet migrated them to version `v2`.
```

> The APIRule v1beta1 API is no longer available either via the Kyma Dashboard.

Reviewer: ```suggestion
The APIRule `v1beta1` API is no longer available via Kyma Dashboard.
```
